### PR TITLE
Use mmap file allocator in dense tensor attribute if huge flag is set in config.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    vespalib::string _dir_name;
+    std::atomic<uint64_t> _generation;
+
+    MmapFileAllocatorFactory();
+    ~MmapFileAllocatorFactory();
+    MmapFileAllocatorFactory(const MmapFileAllocatorFactory &) = delete;
+    MmapFileAllocatorFactory& operator=(const MmapFileAllocatorFactory &) = delete;
+public:
+    void setup(const vespalib::string &dir_name);
+    virtual std::unique_ptr<MemoryAllocator> get_memory_allocator(const vespalib::string& name);
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h:26]: Should we remove _virtual_ here?

Comment (geirst) [inline vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h:26]: Consider rename to _make_memory_allocator()_ to reflect that it is a factory.

File: vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+MmapFileAllocatorFactory::~MmapFileAllocatorFactory()
+{
+}
+
+void
+MmapFileAllocatorFactory::setup(const vespalib::string& dir_name)
+{
+    _dir_name = dir_name;
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.cpp:22]: Do we need locking here?

**Review by geirst — APPROVED** [review]
👍

File: vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h (hunk 1 above)
Comment (toregge) [reply vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h]: Done

Comment (toregge) [reply vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.h]: Done

File: vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.cpp (hunk 2 above)
Comment (toregge) [reply vespalib/src/vespa/vespalib/util/mmap_file_allocator_factory.cpp:22]: It should not be needed as long as the method is not misused.

**Review by geirst — APPROVED** [review]
👍

**Comment by geirst** [comment]
Failing screwdriver build is not related:
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:2.5.2:install (default-install) on project bundle-plugin: Failed to install metadata com.yahoo.vespa/maven-metadata.xml: Error updating group repository metadata: in epilog non whitespace content is not allowed but got / (position: END_TAG seen ...</metadata>\n/... @16:2) -> [Help 1]`

**Pull request merged**